test/e2e/framework/log.go

```diff
@@ -94,13 +96,15 @@ func PrunedStack(skip int) string {
 	if len(stack) > 2*skip {
 		stack = stack[2*skip:]
 	}
-	prunedStack := []string{}
+	var prunedStack strings.Builder
 	for i := 0; i < len(stack)/2; i++ {
 		// We filter out based on the source code file name.
-		if !codeFilterRE.Match([]byte(stack[i*2+1])) {
-			prunedStack = append(prunedStack, stack[i*2])
-			prunedStack = append(prunedStack, stack[i*2+1])
+		if !codeFilterRE.Match(stack[i*2+1]) {
+			prunedStack.Write(stack[i*2])
+			prunedStack.WriteByte('\n')
+			prunedStack.Write(stack[i*2+1])
+			prunedStack.WriteByte('\n')
```

Comment:
would be interested to see what performance benefits this gives.

Reply:
@neolit123 overall it's 1.5x speedup, which is much less than I anticipated.

I switched to another approach (filtering the slice in-place), which gives me 1.9x.